src/main/java/com/codeborne/selenide/impl/ScreenShotLaboratory.java

```diff
@@ -356,11 +358,7 @@ public String formatScreenShotPath(Driver driver) {
         String name = target.toFile().getName();
         screenshotUrl = driver.config().reportsUrl() + name;
       }
-      try {
-        screenshotUrl = new URL(screenshotUrl).toExternalForm();
-      } catch (MalformedURLException ignore) {
-        // ignored exception
-      }
+      screenshotUrl = normalizeURL(screenshotUrl);
```

Comment:
Should this be removed at all?

It looks like the problem is the URL is combined as string, then it is `normalized`.

Why don't we create the URL via the proper APIs in the first place?

For instance, `screenshotUrl = new URL(.., driver.config().reportsUrl(), ..., name, ...)` and so on.

Reply:
Hi @vlsi! thanks for your review! One moment is... some URLs can contain special characters, for example, the space character, and to make these characters legal they need to be encoded. So the multi-argument URI constructor automatically takes care of the encoding...

Reply:
> So the multi-argument URI constructor automatically takes care of the encoding...

Well, I thought exactly behind those lines, and I wonder why do you need **both** `new URL` and `new URI` at the same time.

Isn't just one of them enough?

Reply:
For example, new URL("http", "ci.org/with space/", "path with spaces/123.html") does not encode it... 

Creating URL first helps me to take its parts and pass it to URI because If I pass the whole string to URI (e.g. new URI("http://ci.org/with space/path with spaces/123.html")) it fails too... )) and I am not 100% sure that it is the best solution... So I hope that I explained why I did it

Reply:
How about `java.net.URLEncoder.encode(reportsURL, java.nio.charset.StandardCharsets.UTF_8.name()) + encode(screenShotPath, StandardCharsets.UTF_8.name())` ?

Reply:
yes, I tried it, and for example "http://ci.org" will result in "http%3A%2F%2Fci.org"

but I am not sure if it correct

Reply:
Oh, you are right.

So what should be the behavior if `reportsURL` does contain `?param=value` part?
Is it forbidden?

Where should `screenShotPath` be appended?

Reply:
Hm, seems it will not work... And previously it did not work too... I think that no one used it :) 

And I don't know if it should work.. for example, if the user provides `http://ci.org/atifacts?param=value` and when we append screenshot path, like `http://ci.org/atifacts/screenshotpath?param=value` it breaks the logic, isn't it?

Reply:
As a solution, I think maybe we can document `reportsUrl` to provide baseUrl without query parameters